lib/src/dialogs/macos_alert_dialog.dart

```diff
@@ -0,0 +1,136 @@
+import 'package:flutter/material.dart';
+import 'package:macos_ui/macos_ui.dart';
+
+/// A macOS-style AlertDialog.
+///
+/// A [MacosAlertDialog] must display an [appIcon], [title], [message],
+/// and [primaryButton].
+class MacosAlertDialog extends StatelessWidget {
+  const MacosAlertDialog({
+    Key? key,
+    required this.appIcon,
+    required this.title,
+    required this.message,
+    required this.primaryButton,
+    this.secondaryButton,
+    this.horizontalActions = true,
+  }) : super(key: key);
+
+  /// This should be your application's icon.
+  ///
+  /// The size of this widget should be 56x56.
+  final Widget appIcon;
+  final Widget title;
+  final Widget message;
+
+  /// The primary action a user can take.
+  ///
+  /// Typically a [PushButton].
+  final Widget primaryButton;
+
+  /// The secondary action a user can take.
+  ///
+  /// Typically a [PushButton].
+  final Widget? secondaryButton;
+
+  /// Determines whether to lay out [primaryButton] and [secondaryButton]
+  /// horizontally or vertically.
+  ///
+  /// Defaults to `true`.
+  final bool? horizontalActions;
+
+  @override
+  Widget build(BuildContext context) {
+    return Dialog(
+      backgroundColor: MacosTheme.brightnessOf(context).isDark
+          ? MacosColors.controlBackgroundColor.darkColor
+          : MacosColors.controlBackgroundColor,
+      shape: RoundedRectangleBorder(
+        borderRadius: BorderRadius.circular(7.0),
+      ),
+      child: ConstrainedBox(
+        constraints: BoxConstraints(
+          maxHeight: 300,
+          maxWidth: 260,
+        ),
+        child: Column(
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            const SizedBox(height: 28),
+            ConstrainedBox(
+              constraints: BoxConstraints(
+                maxHeight: 56,
+                maxWidth: 56,
+              ),
+              child: appIcon,
+            ),
+            const SizedBox(height: 28),
+            title,
```

Comment:
Wrap this in a `DefaultTextStyle` and set `MacosTheme.of(context).typography.headline` to `style`